connector/connect/client/jvm/src/main/scala/org/apache/spark/ml/Estimator.scala

```diff
@@ -0,0 +1,97 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.ml
+
+import scala.annotation.varargs
+
+import org.apache.spark.annotation.Since
+import org.apache.spark.ml.param.{ParamMap, ParamPair}
+import org.apache.spark.sql.Dataset
+
+/**
+ * Abstract class for estimators that fit models to data.
+ */
+abstract class Estimator[M <: Model[M]] extends PipelineStage {
```

Comment:
Q: Why not move this to mllib-common ?

Reply:
Similar question for Model / Transformer / Predictor / Pipeline

Reply:
Similar question for Classifier, ProbabilisticClassifier

Reply:
For meta algorithm, I think we can move all of them to mllib-common, but we can do it in following PR.

Reply:
The reason that I didn't add `Transformer` (as well as its subclasses) into the `mllib-common` is to control the dependency:

```
def transform(dataset: Dataset[_], paramMap: ParamMap): DataFrame
```

`Transformer` depends on `Dataset`, I think it will not be involved in the catalyst common module

cc @hvanhovell

Reply:
Can we change it to
`def transform(dataset: DataFrame, paramMap: ParamMap): DataFrame` ?

Shall we still support Dataset[_] input ?